pydantic/_internal/_generate_schema.py

```diff
@@ -1540,6 +1545,8 @@ def _dataclass_schema(
                 self.defs.definitions[dataclass_ref] = self._post_process_generated_schema(schema)
                 return core_schema.definition_reference_schema(dataclass_ref)
 
+            assert False, 'Unreachable'
```

Comment:
mypy and co. seem to think that the ExitStack may supress some exception and therefore fall though the with statement, causing this function to return None.

Reply:
Please add a comment in the code about this